cognite_toolkit/_cdf_tk/commands/modules.py

```diff
@@ -680,9 +709,62 @@ def add(self, organization_dir: Path) -> None:
             build_env = default.environment.validation_type
 
         existing_module_names = [module.name for module in ModuleResources(organization_dir, build_env).list()]
-        available_packages = Packages().load(self._builtin_modules_path)
-
+        available_packages = self._get_available_packages()
         added_packages = self._select_packages(available_packages, existing_module_names)
 
         download_data = self._get_download_data(added_packages)
         self._create(organization_dir, added_packages, environments, "update", download_data)
+
+    def _get_available_packages(self) -> Packages:
+        """
+        Returns a list of available packages, either from the CDF TOML file or from external libraries if the feature flag is enabled.
+        If the feature flag is not enabled and no libraries are specified, it returns the built-in modules.
+        """
+
+        if Flags.EXTERNAL_LIBRARIES.is_enabled() and CDFToml.libraries:
+            for library_name, library in CDFToml.libraries.items():
+                try:
+                    print(f"[green]Adding library {library_name}[/]")
+                    output_path = self._temp_download_dir / f"{library_name}.zip"
+                    self._download_and_unpack(library.url, output_path)
+                    return Packages().load(output_path.parent)
+                except Exception as e:
+                    print(f"[red]Failed to add library {library_name}: {e}[/red]")
+                    raise
+            # If no libraries are specified or the flag is not enabled, load the built-in modules
+        return Packages.load(self._builtin_modules_path)
```

Comment:
Small suggestion, just for readability.
```suggestion
            # If no libraries are specified or the flag is not enabled, load the built-in modules
        else:
            return Packages.load(self._builtin_modules_path)
```